src/main/java/org/sagebionetworks/web/client/widget/user/UserBadgeViewImpl.java

```diff
@@ -187,6 +187,18 @@ public void setHref(String href) {
 	public void clearHref() {
 		usernameLink.setHref(HasHref.EMPTY_HREF);
 	}
+	
+	@Override
+	public void openNewWindow(String url) {
+		DisplayUtils.newWindow(url, "_blank", "");
+		
+	}
+	
+	@Override
+	public void setOpenNewWindow(boolean value) {
+		usernameLink.setTarget(value ? "_blank" : "");
```

Comment:
Please move this logic (the ternary operator is really an if/else) to the presenter and test.